haas/cli.py

```diff
@@ -130,9 +175,33 @@ def setup_http_client():
         http_client = RequestsHTTPClient()
         http_client.auth = (basic_username, basic_password)
         return
-    # Fall back to no authentication:
-    http_client = RequestsHTTPClient()
-
+    # Next try keystone:
+    try:
+        from keystoneauth1.identity import v3
+        from keystoneauth1 import session
+        os_auth_url = os.getenv('OS_AUTH_URL')
+        os_password = os.getenv('OS_PASSWORD')
+        os_username = os.getenv('OS_USERNAME')
+        os_user_domain_id = os.getenv('OS_USER_DOMAIN_ID') or 'Default'
+        os_project_name = os.getenv('OS_PROJECT_NAME')
+        os_project_domain_id = os.getenv('OS_PROJECT_DOMAIN_ID') or 'Default'
+        if None in (os_auth_url, os_username, os_password, os_project_name):
+            # XXX: This error kinda makes sense, but I (zenhack) am somewhat
+            # dissatisfied.
+            raise KeyError
+        auth = v3.Password(auth_url=os_auth_url,
+                           username=os_username,
+                           password=os_password,
+                           project_name=os_project_name,
+                           user_domain_id=os_user_domain_id,
+                           project_domain_id=os_project_domain_id)
+        sess = session.Session(auth=auth)
+        http_client = KeystoneHTTPClient(sess)
+        return
+    except (ImportError, KeyError):
+        pass
```

Comment:
It's a bit trickier than that, because we'd have to refactor the way HTTPClient works, or do the lookup separately; currently it expects a fully qualified URL, I'm sure there's a solution that isn't too bad, but I'd rather hash it out as we're desiging the client library/refactoring the CLI.